### ISSUE TITLE
Visible delay when opening new tabs while system is under load

### ISSUE THREAD

**Issue by theultramage** [tsuliwaensis/New-Tabs-at-End-3000#2]
I'm noticing that when my (underpowered) system is under high load, it takes up to 2 seconds before the newly created tab gets moved to the end. I see the active spinner, and sometimes I also get to see the page url. It is pretty distracting when it happens.

I have examined the differences between v2 and v3 and concluded that the cause is the switch from background:page to background:service_worker.

**Comment by tsuliwaensis** [comment]
The switch from `background:page` to `background:service_worker` was necessary for Manifest v3.

**Comment by theultramage** [comment]
Well yea, obviously, but knowing that doesn't make it any less unpleasant to use. It was so bad that I resorted to patching the v2 manifest back in and ditching the worker. I have not re-tested it since. It could be useful to check if Chrome added a way to control the process priority of the worker. Since it controls the realtime top-level UI, it needs to have highest priority to have a smooth UI experience, but when I tried it back then it felt like below-normal. The v2 version acts instantly even when the host is under heavy hdd and cpu load.
